module/rdpClientCon.c

```diff
@@ -2195,30 +2204,114 @@ rdpDeferredUpdateCallback(OsTimerPtr timer, CARD32 now, pointer arg)
            clientCon->rdp_width, clientCon->rdp_height, clientCon->rdp_Bpp,
            id.width, id.height));
     clientCon->updateScheduled = FALSE;
-    rects = 0;
-    num_rects = 0;
-    LLOGLN(10, ("rdpDeferredUpdateCallback: capture_code %d",
-           clientCon->client_info.capture_code));
-    if (rdpCapture(clientCon, clientCon->dirtyRegion, &rects, &num_rects,
-                   id.pixels, clientCon->cap_left, clientCon->cap_top,
-                   id.width, id.height,
-                   id.lineBytes, XRDP_a8r8g8b8, id.shmem_pixels,
-                   clientCon->cap_width, clientCon->cap_height,
-                   clientCon->cap_stride_bytes,
-                   clientCon->rdp_format, clientCon->client_info.capture_code))
-    {
-        LLOGLN(10, ("rdpDeferredUpdateCallback: num_rects %d", num_rects));
-        rdpClientConSendPaintRectShmEx(clientCon->dev, clientCon, &id,
-                                       clientCon->dirtyRegion,
-                                       rects, num_rects);
-        free(rects);
+    if (clientCon->dev->monitorCount < 1)
```

Comment:
Here's a design pattern I learned from complex projects. Always code for the hardest case. That gives good input for design. Then port to easier cases. Keep additional logic outside the original code by using short functions and macros.